tools/devops/automation/templates/release/publish-nugets.yml

```
@@ -0,0 +1,55 @@
+# all steps that are required to publish the nugets, the template takes the parameters to decide in which feed should the nugets
+# be published
+steps:
+
+# based on the build, decide if we are used a feed or the other and export it
+- bash: |
+    set -x
+    set -e
+    make
+    var=$(make -C $(Build.SourcesDirectory)/xamarin-macios/tools/devops print-variable VARIABLE=$VAR_NAME)
+    var=${var#*=}
+    echo "##vso[task.setvariable variable=Credential]$var"
+  displayName: 'Retrieve nuget feed'
+  name: ConfigureNuget
+  env:
+    ${{ if not(contains (variables['Build.DefinitionName'], 'private')) }}:
+      VAR_NAME=NUGET_PUBLIC_FEED
+    ${{ if contains (variables['Build.DefinitionName'], 'private') }}:
+      VAR_NAME=NUGET_PRIVATE_FEED
+
+- task: NuGetAuthenticate@0
+  displayName: 'Authenticate dotnet-tool feed'
+  inputs:
+    nuGetServiceConnections: $(ConfigureNuget.Credential)
+  condition: contains (variables['Build.Repository.ID'], 'private')
```

@mandel-macaque should we switch this to use `Build.DefinitionName`?

```suggestion
  condition: contains (variables['Build.DefinitionName'], 'private')
```

---

yes, I missed this one. nice catch